logger.go

```diff
@@ -12,11 +13,28 @@ import (
 
 type Logger = *zap.SugaredLogger
 
-var Global Logger
+var (
+	globalMu     sync.RWMutex
+	globalLogger = newDefaultLogger()
+)
+
+func newDefaultLogger() Logger {
+	return zap.Must(NewDevelopmentLoggerConfig().Build()).Sugar()
+}
+
+func ReplaceGloabl(logger Logger) {
```

Comment:
can you have this return the last logger?

Reply:
and/or an undo func like https://github.com/viamrobotics/goutils/blob/37181fad6d7d4f8d78fd15b6e22216a8502c32ea/artifact/google_storage_store.go#L35